app/controllers/api/api_controller.rb

```diff
@@ -0,0 +1,59 @@
+# frozen_string_literal: true
+
+module Api
+  class ApiController < ApplicationController
+    skip_before_action :verify_authenticity_token
+
+    before_action :set_map
+
+    around_action :handle_exceptions
+
+    private
+
+    def handle_exceptions
+      begin
+        yield
+      rescue ActiveRecord::RecordNotFound => e
+        @status = 404
+        @error_type = e.class.to_s
+        # @message = 'Record not found'
+      rescue ArgumentError => e
+        @status = 422
+        @error_type = e.class.to_s
+      rescue StandardError => e
+        @status = 500
+        @error_type = e.class.to_s
+      end
+
+      return if e.instance_of?(NilClass)
```

Comment:
I checked: `e` is only `nil` if none of the errors were rescued. Therefore, we need to keep this line in (otherwise it always renders the next line).

Reply:
Ah it's a guard clause. Looks like we both didn't read it as such :D Wouldn't it be more readable to have a "if error then render error" block instead?

Reply:
No, I did read it as a guard clause 😄 I'd prefer to leave it as a guard clause (Rubocop, too)